src/nvim/option.c

```diff
@@ -844,7 +844,7 @@ set_option_default (
       if (options[opt_idx].indir == PV_SCROLL)
         win_comp_scroll(curwin);
       else {
-        *(intptr_t *)varp = (intptr_t)options[opt_idx].def_val[dvi];
+        *(long *)varp = (long)options[opt_idx].def_val[dvi];
```

Comment:
varp is char_u*, long value will corrupt ??

It's difficult to understand.

Reply:
varp is `union {char_u **strp; int *boolp; long *nump;}`, `char_u*` is legacy because Bram for some reason does not like using `void*` or unions in a number of places: depending on option type it is `int*`, `long*` or `char_u**`, not sure how `intptr_t*` managed to appear here.

Reply:
Done